### PR TITLE
[DOCS] Dynamic Sidebar in progress

### LINKED ISSUE #9916: [Docs] Update the dynamically built toc sidebar of docs
---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 📚 [Instructions for contributing to documentation](https://docs.meshery.io/project/contributing/contributing-docs)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by iArchitSharma**
**Notes for Reviewers**

This PR fixes #9916 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 4c5a5c759807b36d538ca851347f73395dab9e86 at: https://65a2b70f48eafcd3bfa3a2ac--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
Let's be sure to reuse as described in https://github.com/meshery/meshery/issues/9916#issuecomment-1891095858

**Comment by l5io** [comment]
🚀 Preview for commit 255548e359bf778dc4044fbef18506e9bbbed592 at: https://65b4df2341423d7d8e50797d--meshery-docs-preview.netlify.app

**Comment by vishalvivekm** [comment]
@iArchitSharma merge conflicts aren't allowing preview to build. Please resolve these.

**Comment by iArchitSharma** [comment]
@leecalcote
@vishalvivekm, and I are considering a different approach, such as building toc.yml using a script like Go, which runs only once. @sudhanshutech mentioned something about hooks, and that's also something we are looking into.

**Comment by l5io** [comment]
🚀 Preview for commit 4c3bdef05ece2fa5a4b12fca6c1a74733831cce5 at: https://65dc6b1805596ca508ae381d--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit b9d49eea6b55e773cba196ec702a9f7d1b6cf9c0 at: https://65dc708905596ca954ae3a71--meshery-docs-preview.netlify.app

File: docs/pages/project/releases.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ permalink: project/releases
 abstract: List of released Meshery versions and their release notes.
 redirect_from: project/releases/
 language: en
-type: project
+type: Reference
```
Comment (vishalvivekm) [inline docs/pages/project/releases.md:8]: type=project is being used on homepage to make the list: https://github.com/meshery/meshery/blob/fe72c24fd686ce53b739daf81351c59bee385493/docs/pages/index.md?plain=1#L347

File: docs/pages/concepts/architectural/architecture.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ redirect_from: architecture
 type: components
 abstract: overview of different individual components of Meshery architecture and how they interact as a system.
 language: en
-list: include
+list: exclude
```
[the remark this answers is not in the document]
Comment (iArchitSharma) [reply docs/pages/concepts/architectural/architecture.md:9]: if we include this page than this gets listed along with its child pages and its already being listed in home page too 
image

image

File: docs/pages/project/releases.md (hunk 1 above)
Comment (iArchitSharma) [reply docs/pages/project/releases.md:8]: oh i see, let me see how we can fix this

File: docs/pages/concepts/architectural/architecture.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (iArchitSharma) [reply docs/pages/concepts/architectural/architecture.md:9]: logical page was already excluded

**Comment by l5io** [comment]
🚀 Preview for commit a3058f74b5a52ee1f5481c9212210f607f8279a8 at: https://65e5dd337e180d333f26b632--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit a04d2daefe06aa5cd1475da24fe3eacc8ade4f2d at: https://65e5e01c11e13934429682d1--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 1c46d8d8448b440d03ecc3ba18c1375872d4db7e at: https://65e7e03a314708a9a1d55968--meshery-docs-preview.netlify.app

**Comment by vishalvivekm** [comment]
@iArchitSharma will you see why Tutorials isn't begin listed in sidebar under Guides(all) given it exists ? 

ref: https://github.com/meshery/meshery/blob/master/docs/pages/guides/tutorials/index.md

**Comment by vishalvivekm** [comment]
#### current preview:


note: we're missing the border-bottom which distinguishes the sections in TOC.
<hr/>

### live docs:

**Comment by l5io** [comment]
🚀 Preview for commit 40638d930637e6a18e9d90fa926e093f97cb078c at: https://65e8800f11dd1800993b23c6--meshery-docs-preview.netlify.app

**Comment by iArchitSharma** [comment]
**Change in build time:**
Dynamic sidebar first serve - 165.426 seconds
Dynamic sidebar second serve - 24.162 seconds

Original first serve - 106.149 seconds
Original second serve - 16.465 seconds

**Comment by iArchitSharma** [comment]
**Logic behind pages:**
child page -> type, category=nil
grandchild -> type = same as child, category = type-child of child page

**Pull request closed without merging**